### DISCUSSION TITLE
Feature request: warning for missing "import 'twin.macro';"

### DISCUSSION THREAD

**Discussion by matepaiva** [ben-rogerson/twin.macro#437] [Ideas]
Hi all!

In most of my cases, I use the `tw` directive within an HTML tag. The directive does not use any imported lib, but it depends on having an absolute import on every page that we use it. Example:

```jsx
import 'twin.macro';

export default function Footer() {
  return <footer tw="bg-almost-black text-gray-300 px-6 lg:px-8 py-6 md:py-12">hello</footer>;
```

The problem is: sometimes we forget to add the `twin.macro` during a refactor.

I was wondering if we could apply a eslint rule or a warning when the file has a `tw` directive but no `twin.macro` import. It's like the warning eslint shows when you use jsx without importing the `React` lib.

Do you think it is possible?

Thank you

**Comment by ben-rogerson** [comment]
Hey Matheus,

If you're only after the tw prop you can avoid the import altogether with [babel-plugin-twin](https://github.com/ben-rogerson/babel-plugin-twin).

[I'm keen on adding a "tw hasn't been imported" warning](https://github.com/ben-rogerson/babel-plugin-twin/issues/1) too, but it'll only be possible with babel-plugin-twin running in your project.

**Reply by ben-rogerson** [reply]
+ I'm also keen to explore a "eslint-plugin-twin" too

**Comment by matepaiva** [comment]
Thank you, @ben-rogerson! It works like a charm for my use case!
